File: backend/maven/src/main/java/com/dreamteam/atelier/model/domain/UserAddress.java
package com.dreamteam.atelier.model.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;


import java.util.UUID;

/**
 * Базовый тип описывающий сущность - адрес пользователя
 */
@Document(collection = "usersAddresses")
@TypeAlias(value = "userAddress")
@Data
@EqualsAndHashCode
@RequiredArgsConstructor
public class UserAddress {

    @Id
    private String id;
    @Indexed(unique = true)
    private UUID uuid;
    private String region;
    private String city;
    private String street;
    private String house;
    private String flat;
    private String description; //дополнительное описание

}

File: backend/nodejs/src/data/model/user-address.ts
import { Document, Schema, model } from 'mongoose';

export interface IUserAddress {
    region: string;
    city: string;
    street: string;
    house: string;
    floor?: string;
    flat?: string;
    description?: string;
}

export interface IUserAddressDocument extends IUserAddress, Document {
}

export const UserAddressSchema = new Schema({
    region: {
        type: String,
        required: true
    },
    city: {
        type: String,
        required: true
    },
    street: {
        type: String,
        required: true
    },
    house: {
        type: String,
        required: true
    },
    floor: {
        type: String
    },
    flat: {
        type: String
    },
    description: {
        type: String
    }
});

export default model<IUserAddressDocument>('UserAddress', UserAddressSchema);
